tests/store/tracking/test_file_store_logged_model.py

```diff
@@ -0,0 +1,619 @@
+            return_value={"experiment_id": "1"},
+        ),
+    ):
+        with pytest.raises(MlflowException, match=r"Model '1234' metadata is in invalid state"):
+            store.get_logged_model("1234")
+
+
+def test_finalize_logged_model(store):
+    logged_model = store.create_logged_model()
+    assert logged_model.status == str(LoggedModelStatus.PENDING)
+    logged_model_dict = logged_model.to_dictionary()
+    updated_model = store.finalize_logged_model(logged_model.model_id, LoggedModelStatus.READY)
+    assert updated_model.status == str(LoggedModelStatus.READY)
+    updated_model_dict = updated_model.to_dictionary()
+    for k in logged_model_dict:
+        if k not in ["status", "last_updated_timestamp"]:
+            assert logged_model_dict[k] == updated_model_dict[k]
+
+
+def test_finalize_logged_model_errors(store):
+    with pytest.raises(MlflowException, match=r"Model '1234' not found"):
+        store.finalize_logged_model("1234", LoggedModelStatus.READY)
+
+    logged_model = store.create_logged_model()
+    with pytest.raises(MlflowException, match=r"Invalid model status"):
+        store.finalize_logged_model(logged_model.model_id, LoggedModelStatus.UNSPECIFIED)
+
+
+def test_search_logged_models_experiment_ids(store):
+    exp_ids = []
+    for i in range(5):
+        exp_id = store.create_experiment(f"test_{i}")
+        store.create_logged_model(exp_id)
+        exp_ids.append(exp_id)
+    assert len(store.search_logged_models(experiment_ids=exp_ids)) == 5
+    for exp_id in exp_ids:
+        assert len(store.search_logged_models(experiment_ids=[exp_id])) == 1
+    assert len(store.search_logged_models(experiment_ids=[])) == 0
+    assert len(store.search_logged_models(experiment_ids=exp_ids + ["1234"])) == 5
+
+
+def test_search_logged_models_filter_string(store):
+    exp_id = store.create_experiment("test")
+    run_ids = []
+    logged_models = []
+    for i in range(5):
+        run_ids.append(store.create_run(exp_id, "user", 0, [], f"test_run_{i}").info.run_id)
+        logged_models.append(
+            store.create_logged_model(exp_id, source_run_id=run_ids[-1], model_type="test")
+        )
+        # make sure the creation_timestamp is different
+        time.sleep(0.001)
+
+    # model_id
+    # TODO: do we need to support IN & NOT IN?
+    for model in logged_models:
+        models = store.search_logged_models(
+            experiment_ids=[exp_id], filter_string=f"model_id='{model.model_id}'"
+        )
+        assert len(models) == 1
+        assert models[0].to_dictionary() == model.to_dictionary()
+
+    model = logged_models[0]
+    models = store.search_logged_models(
+        experiment_ids=[exp_id], filter_string=f"model_id!='{model.model_id}'"
+    )
+    assert len(models) == 4
+
+    models = store.search_logged_models(
+        experiment_ids=[exp_id], filter_string=f"model_id LIKE '{model.model_id}'"
+    )
+    assert len(models) == 1
```

Comment:
Can we check the element instead of the length? `models` might looks like `[None]`. This test should fail if so.